internal/testing/validation/postgrescluster_test.go

```diff
@@ -519,3 +520,101 @@ func TestPostgresUserOptions(t *testing.T) {
 		assert.NilError(t, cc.Create(ctx, cluster, client.DryRunAll))
 	})
 }
+
+func TestPostgresUserInterfaceAcrossVersions(t *testing.T) {
+	ctx := context.Background()
+	cc := require.Kubernetes(t)
+	t.Parallel()
+
+	namespace := require.Namespace(t, cc)
+
+	base := v1beta1.NewPostgresCluster()
+	// Start with a bunch of required fields.
+	base.Namespace = namespace.Name
+	base.Name = "postgres-pgadmin"
+	require.UnmarshalInto(t, &base.Spec, `{
+		userInterface: {
+			pgAdmin: {
+				dataVolumeClaimSpec: {
+					accessModes: [ReadWriteOnce],
+					resources: { requests: { storage: 1Mi } },
+				},
+			},
+		},
+		postgresVersion: 16,
+		backups: {
```

Comment:
🤔 I remember now that I tried turning `backups` into a pointer so we could omit it in these tests, and it sent me down some long hole.